### PR TITLE
Add Vega 2 renderer

### PR DESCRIPTION
**Pull request by domoritz**
Companion to https://github.com/jupyterlab/jupyterlab/pull/4107

Thanks to @tongshuangwu and @NeelMohapatra

### CODE REVIEW

File: packages/vega2-extension/src/index.ts [hunk 1]
```diff
@@ -0,0 +1,371 @@
+/*-----------------------------------------------------------------------------
+| Copyright (c) Jupyter Development Team.
+| Distributed under the terms of the Modified BSD License.
+|----------------------------------------------------------------------------*/
+
+import {
+<<<<<<< Updated upstream
```
Comment (domoritz) [inline packages/vega2-extension/src/index.ts]: another merge marker

**Comment by domoritz** [comment]
Thanks!

File: packages/vega2-extension/src/index.ts [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+    // Handle things related to the MIME type.
+    let mimeType = this._mimeType = options.mimeType;
+    if (mimeType === VEGA_MIME_TYPE) {
+      this.addClass(VEGA_CLASS);
+    } else {
+      this.addClass(VEGALITE_CLASS);
+    }
+  }
+
+  // /**
```
Comment (domoritz) [inline packages/vega2-extension/src/index.ts]: Can we delete this?

File: packages/vega2-extension/src/index.ts [hunk 3]
```diff
@@ -0,0 +1,246 @@
...
+    displayName: 'Vega-Lite 1 File',
+    extensions: ['.vl', '.vl.json', '.vegalite'],
+    iconClass: 'jp-MaterialIcon jp-VegaIcon',
+  }]
+};
+
+export default extension;
+
+
+// /**
```
Comment (domoritz) [inline packages/vega2-extension/src/index.ts]: Can we delete this?

File: packages/vega2-extension/tsconfig.json [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+    "noImplicitAny": false,
+    "noEmitOnError": true,
+    "noUnusedLocals": true,
+    "module": "commonjs",
+    "moduleResolution": "node",
+    "target": "ES5",
+    "outDir": "./lib",
+    "lib": ["ES5", "ES2015.Promise", "DOM", "ES2015.Collection"]
+  },
+  "include": ["src/*"]
```
Comment (domoritz) [inline packages/vega2-extension/tsconfig.json]: I think if you add an include of "typings/vega-embed.d.ts", you can change the implicitAny to false.

Comment (tongshuangwu) [reply packages/vega2-extension/tsconfig.json]: Let me try XD

**Comment by domoritz** [comment]
@gnestor @blink1073 This is ready to go except for the issue I noted in https://github.com/jupyterlab/jupyterlab/pull/4107#issuecomment-371422228. I'd love your advice.

File: packages/vega3-extension/README.md [hunk 5]
```diff
@@ -16,7 +16,7 @@ To render Vega 3 or Vega-lite 2 output in IPython:
 from IPython.display import display
 
 display({
-    "application/vnd.vega-lite.v2+json": {
+    "application/vnd.vegalite.v2+json": {
```
Comment (domoritz) [inline packages/vega3-extension/README.md]: I believe this was a bug in the readme for Vega 3.

File: postBuild [hunk 6]
```diff
@@ -1,3 +1,3 @@
 #!/bin/bash
 
-jupyter labextension install @jupyterlab/fasta-extension @jupyterlab/geojson-extension @jupyterlab/katex-extension @jupyterlab/plotly-extension @jupyterlab/vega3-extension
+jupyter labextension install @jupyterlab/fasta-extension @jupyterlab/geojson-extension @jupyterlab/katex-extension @jupyterlab/plotly-extension @jupyterlab/vega2-extension @jupyterlab/vega3-extension
```
Comment (gnestor) [inline postBuild]: If we're moving vega3-extension to jupyterlab core we shouldn't need to to install it here

Comment (domoritz) [reply postBuild]: Let me remove the extension from here then.

**Comment by domoritz** [comment]
Done. I removed the Vega 3 extension completely.

**Comment by gnestor** [comment]
LGTM!!

**Pull request merged**